File: tutorial01/java-gradle/src/main/java/org/luwrain/app/example/Extension.java
package org.luwrain.app.example;

import org.luwrain.core.*;

public class Extension extends EmptyExtension {
@Override
public Command[] getCommands(Luwrain luwrain) {
return new Command[] {new SimpleShortcutCommand("example")};
}

@Override
public ExtensionObject[] getExtObjects(Luwrain luwrain) {
return new ExtensionObject[] {new SimpleShortcut("example", App.class)};
}

@Override
public void i18nExtension(Luwrain luwrain, org.luwrain.i18n.I18nExtension i18n) {
i18n.addCommandTitle("en", "example", "example 1");
	i18n.addCommandTitle("ru", "example", "Пример 1");
	i18n.addStrings("en", Strings.NAME, new Strings());
	i18n.addStrings("ru", Strings.NAME, new Strings());
}

}

File: tutorial01/kotlin/src/main/kotlin/org/luwrain/app/tutorial01/Extension.kt
package org.luwrain.app.vk2

import org.luwrain.base.*
import org.luwrain.core.*
import org.luwrain.core.extensions.*

class Extension: EmptyExtension() {
    override fun getCommands(luwrain: Luwrain): Array<Command> {
	return arrayOf(SimpleShortcutCommand("vk2"))
    }

    override fun getExtObjects(luwrain: Luwrain): Array<ExtensionObject> {
	return arrayOf(SimpleShortcut("vk2", App::class.java))
    }

override fun i18nExtension(luwrain: Luwrain, i18n: org.luwrain.i18n.I18nExtension) {
		i18n.addCommandTitle("en", "vk2", "VKontakte")
	i18n.addCommandTitle("ru", "vk2", "ВКонтакте")
	    i18n.addStrings("ru", "luwrain.vk2", Strings())
}

}
